Makefile

```diff
@@ -127,18 +129,29 @@ prepush: check-nginx
 .PHONY: release
 release: all deps docs $(SMF_MANIFESTS) check-nginx
 	@echo "Building $(RELEASE_TARBALL)"
+	@$(ROOT)/build/node/bin/node ./node_modules/.bin/kthxbai
 	@mkdir -p $(RELSTAGEDIR)/root/opt/smartdc/mako
 	@mkdir -p $(RELSTAGEDIR)/root/opt/smartdc/boot
 	@mkdir -p $(RELSTAGEDIR)/site
 	@touch $(RELSTAGEDIR)/site/.do-not-delete-me
 	cp -r $(ROOT)/bin \
 	    $(ROOT)/boot \
-	    $(ROOT)/build \
 	    $(ROOT)/build/nginx \
+	    $(ROOT)/lib \
 	    $(ROOT)/node_modules \
 	    $(ROOT)/sapi_manifests \
 	    $(ROOT)/smf \
+	    $(ROOT)/test \
 	    $(RELSTAGEDIR)/root/opt/smartdc/mako/
+	mkdir -p $(RELSTAGEDIR)/root/opt/smartdc/mako/build
+	cp -r $(ROOT)/build/scripts $(RELSTAGEDIR)/root/opt/smartdc/mako/build/
+	mkdir -p $(RELSTAGEDIR)/root/opt/smartdc/mako/build/node/bin \
+		$(RELSTAGEDIR)/root/opt/smartdc/mako/build/node/lib
+	cp -r $(ROOT)/build/node/share $(RELSTAGEDIR)/root/opt/smartdc/mako/build/node/
+	cp -r $(ROOT)/build/node/lib/dtrace $(RELSTAGEDIR)/root/opt/smartdc/mako/build/node/lib/
+	cp -r $(ROOT)/build/node/share $(RELSTAGEDIR)/root/opt/smartdc/mako/build/node/
+	cp $(ROOT)/build/node/bin/node $(RELSTAGEDIR)/root/opt/smartdc/mako/build/node/bin/
+	chmod 755 $(RELSTAGEDIR)/root/opt/smartdc/mako/build/node/bin/node
```

Comment:
FWIW, sdc-imgapi and some others do this for setting up build/node in their `make release` target:

```
	cp -PR \
		$(TOP)/build/node \
		$(RELSTAGEDIR)/root/opt/smartdc/$(NAME)/build/
	# Trim node
	rm -rf \
		$(RELSTAGEDIR)/root/opt/smartdc/$(NAME)/build/node/bin/npm \
		$(RELSTAGEDIR)/root/opt/smartdc/$(NAME)/build/node/lib/node_modules \
		$(RELSTAGEDIR)/root/opt/smartdc/$(NAME)/build/node/include \
		$(RELSTAGEDIR)/root/opt/smartdc/$(NAME)/build/node/share
```

AFAIK the share, include and lib/node_modules dirs are not needed (at least imgapi and those others are working fine). This has the benefit of droping ~20MB of npm's node_modules deps.

Reply:
The code here already doesn't copy in the node modules in the first place. In a build this is what's currently left:

```
build/node
build/node/share
build/node/share/systemtap
build/node/share/systemtap/tapset
build/node/share/systemtap/tapset/node.stp
build/node/share/doc
build/node/share/doc/node
build/node/share/doc/node/lldbinit
build/node/share/doc/node/gdbinit
build/node/share/doc/node/lldb_commands.py
build/node/share/man
build/node/share/man/man1
build/node/share/man/man1/node.1
build/node/lib
build/node/lib/dtrace
build/node/lib/dtrace/node.d
build/node/bin
build/node/bin/node
```

You're right that we shouldn't need share. I'll remove that.